src/docfx/lib/path/FilePath.cs

```diff
@@ -37,31 +36,72 @@ internal class FilePath : IEquatable<FilePath>, IComparable<FilePath>
         /// </summary>
         public bool IsGitCommit { get; }
 
-        public FilePath(string path, FileOrigin origin = FileOrigin.Default)
-        {
-            Debug.Assert(origin != FileOrigin.Dependency);
+        /// <summary>
+        /// Gets the original file path specified in source map.
+        /// </summary>
+        public PathString? OriginalPath { get; }
 
+        /// <summary>
+        /// Creates an unknown file path.
+        /// </summary>
+        public FilePath(string path)
```

Comment:
Is it possible to consolidate the constructors? Then I would not need to think about which one to choose for different origin...

Reply:
I'm consolidating it here so that each type goes to a different constructor, the reason is that the required field of each type is different.